From ac6c76158eb2c1e9888176e80cad60c82db4be7d Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Fri, 12 Sep 2014 01:56:57 -0400
Subject: [drm] dp-mst-helper: Don't use uninitialized fields of the sideband message header

Message-id: <1410487019-24661-20-git-send-email-airlied@redhat.com>
Patchwork-id: 93931
O-Subject: [RHEL7.1 BZ1140440 PATCH 19/21] drm/dp-mst-helper: Don't use uninitialized fields of the sideband message header
Bugzilla: 1140440
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Rob Clark <rclark@redhat.com>

From: Damien Lespiau <damien.lespiau@intel.com>

bugzilla: 1140440
upstream: bf3719c04ee3322b0f6d13c078e3ca4c3b7e3ead

We could be using uninitialized fields of the header in
drm_dp_encode_sideband_msg_hdr(), for instance hdr->somt is set to 1 in
the first patcket but never set to 0 otherwise.

Always clear the header at the start then.

Signed-off-by: Damien Lespiau <damien.lespiau@intel.com>
Signed-off-by: Daniel Vetter <daniel.vetter@ffwll.ch>
Signed-off-by: Dave Airlie <airlied@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/drm_dp_mst_topology.c b/drivers/gpu/drm/drm_dp_mst_topology.c
index fbe7f04..6bc16d7 100644
--- a/drivers/gpu/drm/drm_dp_mst_topology.c
+++ b/drivers/gpu/drm/drm_dp_mst_topology.c
@@ -1292,6 +1292,8 @@ static int process_single_tx_qlock(struct drm_dp_mst_topology_mgr *mgr,
  int len, space, idx, tosend;
  int ret;
 
+ memset(&hdr, 0, sizeof(struct drm_dp_sideband_msg_hdr));
+
  if (txmsg->state == DRM_DP_SIDEBAND_TX_QUEUED) {
   txmsg->seqno = -1;
   txmsg->state = DRM_DP_SIDEBAND_TX_START_SEND;
-- 
1.7.1